roles/common/tasks/get_authorization_token.yml

```diff
@@ -71,16 +73,16 @@
   no_log: "{{mask_secrets|bool}}"
   when:
     - cluster_id_source | default('erp') == 'erp'
-    - auth_mode == 'mtls'
+    - auth_mode == 'mtls' or cert_auth_only_enabled|bool
 
 - name: Get Authorization Token
   set_fact:
     authorization_token: >-
-      {%- if oauth|bool -%}
+      {%- if oauth|bool and not cert_auth_only_enabled|bool -%}
         {{ (auth_token_oauth.content | from_json).access_token }}
-      {%- elif not oauth|bool and auth_mode != 'mtls' -%}
+      {%- elif not oauth|bool and auth_mode != 'mtls' and not cert_auth_only_enabled|bool -%}
```

Comment:
does this refer to ldap and ldap with mtls cases?

Reply:
yes both ldap and ldap with mtls are in this case

Reply:
i think it will be better to change the condition to something like `ldap|bool and auth_mode!=mtls` rather than this.

Reply:
do u mean like from wherever we call get_authorization_token.yml we define `oauth: "{{ oauth_enabled }}"
so similarly also add another variable say `ldap: "{{ 'ldap' in auth_mode }}"` and inside this file change the condition to `"{{ ldap|bool and not cert_auth_only_enabled|bool }}"` ?

Reply:
yes, earlier we only had oauth and non oauth. Now there are multiple combinations due to mtls only and mtls dual. It makes sense to add ldap var now similar to oauth. Adding to this, how are we using `auth_mode` variable in this playbook without passing it during the get_authorization_token call? That would mean it will always take the auth_mode for kafka and not component level, which are used in Connect and SR health checks

Reply:
sorry ignore the last part, just noticed it's anyways making mds api call

Reply:
My idea was to use 2 things.
1. oauth variable which has the information regarding component's oauth as well, As in health checks it define oauth: "{{compoennt_oauth_enabled}}" before calling get token. 
2. auth_mode == mtls. When this happens auth_mode in each component will also be mtls, it cant be different as MDS doesnt have ability to support oauth/ldap credentials thus the component also should not. 
3. Then using these 2's negeation I am writing the condition for ldap. so when its not oauth and not mtls then its ldap.
Therefore I believe this works correctly but maybe the conditions are now a bit complex and could be simplified by adding one more variable

Reply:
yes, agreed with the functioning, it's just a bit messy. Though this can be taken up later as an improvement since it is not a breaking change/priority